ecommerce/backend/API/views/vendor.py

```diff
@@ -14,3 +14,20 @@ def vendor_details(request,vendor_id):
     if vendor is None:
         return Response(status=status.HTTP_400_BAD_REQUEST)
     return Response({'title':vendor.title,'image_url':vendor.image_url,'first_name':vendor.first_name,'last_name':vendor.last_name,'last_name':vendor.last_name,'total_rating':vendor.total_rating,'rating_count':vendor.rating_count,'description':vendor.description})
+
+@api_view(['PUT'])
+@permission_classes([permissions.IsVendorUser]) 
+def vendor_profile(request):
+    vendor = User.objects.filter(username=request.user).first()
+    img_array = base64.b64decode(request.data["image"])
```

Comment:
You should check if no image file is sent from frontend. 
For example, vendor already have a profile picture and wants to only update his/her description. In this case, this code forces vendor to send same picture again to backend.

If image is null, we should not change image_url field.